compile error when installing from opam

Using the 4.02.3 switch and running `opam install jingoo`, I get the following output:

```
$ opam install jingoo
The following actions will be performed:
  - install jingoo 1.2.10

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default] https://opam.ocaml.org/archives/jingoo.1.2.10+opam.tar.gz downloaded

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of jingoo failed at "gmake".
Processing  1/1: [jingoo: ocamlfind remove]
#=== ERROR while installing jingoo.1.2.10 =====================================#
# opam-version 1.2.1
# os           freebsd
# command      gmake
# path         /home/vagrant/.opam/4.02.3/build/jingoo.1.2.10
# compiler     4.02.3
# exit-code    2
# env-file     /tmp/jingoo/jingoo-18644-e7966f.env
# stdout-file  /tmp/jingoo/jingoo-18644-e7966f.out
# stderr-file  /tmp/jingoo/jingoo-18644-e7966f.err
### stdout ###
# ocamlfind ocamlopt -g -package unix,dynlink,pcre,batteries -c jg_types.mli jg_types.ml jg_utils.ml jg_stub.mli jg_stub.ml  jg_parser.mli jg_parser.ml jg_lexer.ml  jg_runtime.ml jg_interp.ml  jg_template.mli jg_template.ml
# [...]
# ocamlfind ocamlopt -g -package unix,dynlink,pcre,batteries -o jingoo.cmxa -a 
# ocamlfind ocamlopt -g -thread -c 
# ocamlfind ocamlc -g -thread -c 
# ocamlfind ocamlopt -g -o compiler -linkpkg -package unix,dynlink,pcre,batteries jingoo.cmxa jg_cmdline.ml
# *** Error code 2
# 
# Stop.
# make[1]: stopped in /usr/home/vagrant/.opam/4.02.3/build/jingoo.1.2.10/src
# Makefile:2: recipe for target 'all' failed
### stderr ###
# Warning 3: deprecated: String.create
# [...]
# Use Bytes.create instead.
# File "jg_utils.ml", line 95, characters 14-27:
# Warning 3: deprecated: String.create
# Use Bytes.create instead.
# File "jg_cmdline.ml", line 1:
# Error: No implementations provided for the following modules:
#          Jg_template referenced from jg_cmdline.cmx
#          Jg_types referenced from jg_cmdline.cmx
# gmake: *** [all] Error 1



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  - install jingoo 1.2.10

No changes have been performed
```

--- a/Makefile
+++ b/Makefile
@@ -1,26 +1,25 @@
 all:
-	cd src && make
+	$(MAKE) -C src
 
 byte:
 	cd src && make byte
 
 test:
-	cd tests && make run
+	$(MAKE) -C tests run
 
-reinstall:
-	cd src && make uninstall && make install
+reinstall: uninstall install
 
 install:
-	cd src && make install
+	$(MAKE) -C src install
 
 uninstall:
-	cd src && make uninstall
+	$(MAKE) -C src uninstall
 
 clean:
-	cd src && make clean
-	cd tests && make clean
+	$(MAKE) -C src clean
+	$(MAKE) -C tests clean
 
 rebuild:
-	cd src && make rebuild
+	$(MAKE) -C src rebuild
 
 .PHONY: all test reinstall install uninstall reinstall clean rebuild

--- a/example/Makefile
+++ b/example/Makefile
@@ -25,6 +25,6 @@ clean:
 	rm -f *.cmi *.cmo *.cmx *.o *.cmxs
 	rm -f $(TARGET)
 
-rebuild:
-	make clean
-	make all
+rebuild: clean all
+
+.PHONY: rebuild clean all

--- a/src/Makefile
+++ b/src/Makefile
@@ -55,15 +55,13 @@ install: all
 uninstall:
 	ocamlfind remove jingoo
 
-reinstall:
-	make uninstall
-	make install
+reinstall: uninstall install
 
 clean:
 	rm -f *.cmi *.cmo *.cmx *.cma *.a *.cmxa *.o *.annot *.byte
 	rm -f jg_lexer.ml jg_parser.ml jg_parser.mli
 	rm -f $(COMPILER)
 
-rebuild:
-	make clean
-	make all
+rebuild: clean all
+
+.PHONY: rebuild clean reinstall uninstall install compiler

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -8,7 +8,6 @@ CMPL:=ocamlfind ocamlopt -g
 
 all: test
 
-.PHONY:test
 test: $(SRCS)
 	$(CMPL) -o $@ -package $(PKGS) -linkpkg -I ../src ../src/jingoo.cmxa $(SRCS)
 
@@ -22,3 +21,4 @@ rebuild:
 run: test
 	./test
 
+.PHONY: run rebuild clean all

